Detectors/EMCAL/simulation/src/Digitizer.cxx

```diff
@@ -47,33 +82,43 @@ void Digitizer::process(const std::vector<Hit>& hits, std::vector<Digit>& digits
   mDigits.clear();
   mMCTruthContainer.clear();
 
+  if (mSimulateNoiseDigits) {
+    addNoiseDigits();
+  }
+
   for (auto hit : hits) {
     try {
-      Int_t LabelIndex = mMCTruthContainer.getIndexedSize();
-      Digit digit = hitToDigit(hit, LabelIndex);
-      Int_t id = digit.GetTower();
+      std::vector<Digit> digitvector;
```

Comment:
Should be reusable, i.e. as member

Reply:
Indeed. Repeated allocations (as done by vector) should be avoided.